File: telepy.py
#CLI like interface

import argparse, getopt, os, io, struct, mtproto
from classes.shell import TelepyShell

if __name__ == '__main__':
  parser = argparse.ArgumentParser('telepy',description='Python implementation of telegram API.')
  parser.add_argument('command', nargs='?', choices=['cmd', 'dialog_list', 'contact_list'] + ['chat_' + sub for sub in ['info', 'add_user', 'add_user_to_chat', 'del_user', 'set_photo', 'rename']])
  parser.add_argument('args', nargs='*')

  #for command, args, help in (('info', 1, 'prints info about chat'), ('add_user', 2, 'add user to chat'), ('del_user', 2, 'remove user from chat'), ('set_photo', 1, 'sets group chat photo. Same limits as for profile photos.')):
  #  parser.add_argument('chat_' + command, nargs=args, help=help)
  #parser.add_argument
  args = parser.parse_args()

  if args.command is None:
    TelepyShell().cmdloop()

File: telepy.php
<?php
set_include_path(get_include_path() . PATH_SEPARATOR . dirname(__FILE__) . DIRECTORY_SEPARATOR . 'libpy2php');
require_once ('libpy2php.php');
require_once ('argparse.php');
require_once ('mtproto.php');
require_once ('classes/shell.php');



if (($__name__ == '__main__')) {
    $parser = py2php_kwargs_function_call('argparse::ArgumentParser', ['telepy'], ["description" => 'Python implementation of telegram API.']);
    py2php_kwargs_method_call($parser, 'add_argument', ['command'], ["nargs" => '?', "choices" => (['cmd', 'dialog_list', 'contact_list'] + array_map(function($sub) { return "chat_" . $sub; }, ['info', 'add_user', 'add_user_to_chat', 'del_user', 'set_photo', 'rename'])
    ) ]);
    py2php_kwargs_method_call($parser, 'add_argument', ['args'], ["nargs" => '*']);
    $args = $parser->parse_args();
    if (($args->command == null)) {
        $newTelePyShell = new TelepyShell();
        $newTelePyShell->cmdloop();
    }
}
